docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1735,206 @@ As a response to blocked request AdGuard returns a short video placeholder.
 >
 > Rules with `$mp4` modifier are not supported by AdGuard Content Blocker, AdGuard for iOS and Safari.
 
+### Rule priorities {#rule-priorities}
+
+Each rule has its own priority, which is necessary when several rules match the request and the filtering system needs to select one of them.
+Priority is measured by a natural integer. The higher the priority of a rule, the higher its priority value.
+In the case of a conflict between two rules with the same priority value, there is undefined behavior, depending on the implementation.
+
+#### Priority counting
+
+To calculate priority, modifiers have been divided into several groups. They are arranged in order of priority from lowest to highest. The more a modifier narrows the scope of a rule, the more priority it adds to the total weight of the rule or the more a rule covers requests, the lower its priority. Note that there are modifiers where setting one parameter has a higher priority than setting two or more, for example, for `$domain=example.com|example.org`, because a rule with two domains has a smaller, infinitesimal, effective area, but still larger than a rule with one resolved domain.
+
+The base priority weight of any rule is 1. If the priority is a floating point number, it will be **rounded up** to the smaller integer greater than or equal to the calculated weight.
+
+> **Compatibility with different versions of AdGuard**
+>
+> Numerical priorities for rules are available in tsurlfilter since version `v2.1.0`.
+
+#### 1. Basic modifiers, the presence of each adds 1 to the priority: {#priority-category-1}
+
+[//]: # (Please keep them sorted)
+
+ * [`$app`](#app-modifier) with forbidden applications using `~`,
+ * [`$denyallow`](#denyallow-modifier),
+ * [`$dnsrewrite`](#dnsrewrite-modifier),
+ * [`$domain`](#domain-modifier) with forbidden domains using `~`,
+ * [`$header`](#header-modifier),
+ * [`$match-case`](#match-case-modifier),
+ * [`$method`](#method-modifier),
+ * [`$third-party`](#third-party-modifier),
+ * [`$to`](#to-modifier),
```

Comment:
Shouldn't included/excluded domains matter for `$to`, too? 🤔

Reply:
for `$to` it works like `$denyallow`

Reply:
But `$to` is not equivalent to `$denyallow`: with `$to` the scope of the rule can be limited to a small set of domains (much like with `$domain` or `$from`), which is why I ask if we should apply the same wegiht calculation as for `$domain`. 

And in the case of a `$method` modifier, the situation is similar to content-types, so shouldn't it be treated the same?

Reply:
for `$method` i think it should works like for allowed/restricted content-types
wait for @ameshkov thoughts

Reply:
The point of `$to` is often used to actually extend the rule scope, i.e. to do something like that: `/path$to=example.org|example.com`, this is basically a replacement for a regular expression like this: `/(example.org|example.com)\/path/`

Regarding `$method`, I don't have a particular preference here, either way is okay.

Reply:
@ameshkov approved that `$method` can be evaluated as allowed/restricted content-types